pogom/search.py

```diff
@@ -61,47 +52,61 @@ def send_map_request(api, position):
         log.warning("Uncaught exception when downloading map " + str(e))
         return False
 
+def get_new_coords(init_loc, distance, bearing):
+    """ Given an initial lat/lng, a distance(in kms), and a bearing (degrees),
+    this will calculate the resulting lat/lng coordinates.
+    """ 
+    R = 6378.1 #km radius of the earth
+    bearing = math.radians(bearing)
 
-def generate_location_steps(initial_location, num_steps):
-
-    ring = 1  # Which ring are we on, 0 = center
-    lat_location = initial_location[0]
-    lng_location = initial_location[1]
-
-    yield (initial_location[0], initial_location[1], 0)  # Middle circle
-
-    while ring < num_steps:
-        # Move the location diagonally to top left spot, then start the circle which will end up back here for the next ring
-        # Move Lat north first
-        lat_location += lat_gap_degrees
-        lng_location -= calculate_lng_degrees(lat_location)
+    init_coords = [math.radians(init_loc[0]), math.radians(init_loc[1])] # convert lat/lng to radians
 
-        for direction in range(6):
-            for i in range(ring):
-                if direction == 0:  # Right
-                    lng_location += calculate_lng_degrees(lat_location) * 2
-
-                if direction == 1:  # Right Down
-                    lat_location -= lat_gap_degrees
-                    lng_location += calculate_lng_degrees(lat_location)
+    new_lat = math.asin( math.sin(init_coords[0])*math.cos(distance/R) +
+        math.cos(init_coords[0])*math.sin(distance/R)*math.cos(bearing))
 
-                if direction == 2:  # Left Down
-                    lat_location -= lat_gap_degrees
-                    lng_location -= calculate_lng_degrees(lat_location)
+    new_lon = init_coords[1] + math.atan2(math.sin(bearing)*math.sin(distance/R)*math.cos(init_coords[0]),
+        math.cos(distance/R)-math.sin(init_coords[0])*math.sin(new_lat))
```

Comment:
__[flake8]__
* [E128] continuation line under-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->